wrapanapi/systems/redfish.py

```diff
@@ -168,6 +225,82 @@ def server_inventory(self, physical_server, requested_items, **kwargs):
         return {item: self._server_inventory_available[item](redfish_server)
                 for item in requested_items}
 
+    def rack_stats(self, physical_rack, requested_stats, **kwargs):
+        """
+        Evaluate the requested rack stats at the API server.
+
+        Returns a dictionary of stats and their respective evaluated values.
+
+        Args:
+          physical_rack: representation for the class of this method's caller
+          requested_stats: the statistics to be obtained for the rack
+        """
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._rack_stats_available
+
+        # Get an instance of the requested Redfish rack
+        redfish_rack = self.get_rack(physical_rack.ems_ref)
+
+        return {stat: self._rack_stats_available[stat](redfish_rack)
+                for stat in requested_stats}
```

Comment:
There's a bunch of boilerplate code in those 4 methods :) I'm thinking to refactor into something more like:

```python
def rack_stats(self, ...):
  return evaluate_lambdas(hash_with_lambdas, keys, lambda_args)

def rack_inventory(self, ...):
  return evaluate_lambdas(hash_with_lambdas, keys, lambda_args)

def chassis_stats(self, ...):
  return evaluate_lambdas(hash_with_lambdas, keys, lambda_args)

def chassis_inventory(self, ...):
  return evaluate_lambdas(hash_with_lambdas, keys, lambda_args)
```
(of course, you need to provide correct values for `hash_with_lambdas`, `keys` and `lambda_args`)

Because we're doing the same thing all the time, evaluating correct lambdas... Maybe we could put `evaluate_lambdas` in a Mixin even to make it reusable.

Reply:
Thank you for the suggestion. I created a [ticket](https://github.com/xlab-si/wrapanapi/issues/1) to address this later.